public/app/features/alerting/unified/components/rule-editor/AnnotationsField.tsx

```diff
@@ -81,17 +144,68 @@ const AnnotationsField = () => {
             </div>
           );
         })}
-        <Button
-          className={styles.addAnnotationsButton}
-          icon="plus-circle"
-          type="button"
-          variant="secondary"
-          onClick={() => {
-            append({ key: '', value: '' });
-          }}
+        <Stack direction="row" gap={1}>
+          <Button
+            className={styles.addAnnotationsButton}
+            icon="plus-circle"
+            type="button"
+            variant="secondary"
+            onClick={() => {
+              append({ key: '', value: '' });
+            }}
+          >
+            Add new annotation
+          </Button>
+          <Button type="button" variant="secondary" icon="dashboard" onClick={openDashboardPicker}>
+            Set dashboard and panel
```

Comment:
I think perhaps `Attach dashboard and panel` would be a better verb here? cc @brendamuir